packages/core/src/internal/base/button.base.ts

```diff
@@ -104,26 +119,48 @@ export class CdsBaseButton extends LitElement {
     if (!this.readonly) {
       if (this.disabled) {
         stopEvent(event);
-      } else if (!event.defaultPrevented) {
-        const buttonTemplate = html`<button
-          aria-hidden="true"
-          ?disabled="${this.disabled}"
-          tabindex="-1"
-          style="display: none !important"
-          value="${ifDefined(this.value)}"
-          name="${ifDefined(this.name)}"
-          type="${ifDefined(this.type)}"
-        ></button>`;
-
-        const marker = document.createElement('div');
-        this.appendChild(marker);
-        render(buttonTemplate, marker);
+      } else if (!event.defaultPrevented && this.marker && this.explicitClick) {
+        if (this.markerAttached()) {
+          this.appendChild(this.marker);
+        }
         this.querySelector('button[aria-hidden]')?.dispatchEvent(
           new MouseEvent('click', { relatedTarget: this, composed: true })
         );
-        marker.remove();
+        if (this.type !== 'submit') {
+          this.marker.remove();
+        }
       }
     }
+    // get ready for the next real click check
+    this.explicitClick = false;
+  }
+
+  private createMarker() {
+    const buttonTemplate = html`<button
+      aria-hidden="true"
+      role="presentation"
+      ?disabled="${this.disabled}"
+      tabindex="-1"
+      style="display: none !important"
+      value="${ifDefined(this.value)}"
+      name="${ifDefined(this.name)}"
+      type="${ifDefined(this.type)}"
+    ></button>`;
+
+    const marker = document.createElement('div');
+    // avoid an empty space inside button
+    marker.style.display = 'none';
+    render(buttonTemplate, marker);
+    this.marker = marker;
+  }
+
+  private markerAttached(): boolean {
+    return !!(this.marker && this.marker.parentElement);
+  }
+
+  // A marker significant property is one that should be dynamically bound to a property of the native button
+  private hasMarkerSignificantProperty(props: Map<string, any>): boolean {
+    return props.has('type') || props.has('name') || props.has('value') || props.has('disabled');
```

Comment:
seems like you could have the string values here as an array of strings in a constant (or a private property if preferred) and then loop through them. that way the `or` conditional doesn't get out of hand and you have the strings handy if you need to use them somewhere else in the code.

Reply:
done